addons/mail/static/src/chatter/web/form_compiler.js

```diff
@@ -37,7 +33,7 @@ function compileChatter(node, params) {
     const chatterContainerXml = createElement("t");
     setAttributes(chatterContainerXml, {
         "t-component": "__comp__.mailComponents.Chatter",
-        hasActivities,
+        has_activities: "__comp__.props.archInfo.has_activities",
```

Comment:
From what we discussed, I don't think it's the best way of storing the info conceptually (this is not arch info), but at the same time this is minor and can be changed later if needed, this is a quick and working solution with not a lot of code.